src/v/cluster/tm_stm.cc

```diff
@@ -370,11 +371,11 @@ ss::future<checked<tm_transaction, tm_stm::op_status>> tm_stm::reset_tx_ready(
     co_return co_await update_tx(std::move(tx), expected_term);
 }
 
-checked<tm_transaction, tm_stm::op_status>
-tm_stm::mark_tx_ongoing(kafka::transactional_id tx_id) {
-    auto tx_opt = get_tx(tx_id);
+ss::future<checked<tm_transaction, tm_stm::op_status>>
+tm_stm::mark_tx_ongoing(const kafka::transactional_id& tx_id) {
+    auto tx_opt = co_await get_tx(tx_id);
```

Comment:
is this performance critical? coroutines should take parameters by value, by default. here the caller must ensure that `tx_id` stays alive since it is also accessed down below a the bottom of this method.